### PR TITLE
Add DVC pipeline and GDrive remote

### PR DESCRIPTION
**Pull request by rkargul**

### CODE REVIEW

File: Dockerfile
Comment (rkargul) [inline Dockerfile:1]: I still have to test this

File: README.md
Comment (rkargul) [inline README.md:1]: Lmk if anything is unclear or missing

**Comment by Iodine98** [comment]
@rkargul Could you add a linting job as a Github action? Just to ensure that the ML code is appropriate before building a version as an image.

**Comment by Iodine98** [comment]
> @rkargul Could you add a linting job as a Github action? Just to ensure that the ML code is appropriate before building a version as an image.

I just added linting as a Github Action

**Review by Iodine98 — CHANGES_REQUESTED** [review]
I ran all the linters and the dvc pipeline and it seems to work, but I wonder about the following:

image

- How do we report metrics using DVC?
- How do we approach auditing code quality in the pipeline? Currently the linter is run as a Github action that's triggered by a pull request. 

Once you address these questions, I will approve this review.

**Comment by rkargul** [comment]
> How do we approach auditing code quality in the pipeline?

Apart from running `mllint`, which runs Code Quality (Pylint, Mypy, Black, isort, Bandit), but also the version control (Git and DVC), I am not sure how else we can audit code quality in the pipeline. Any ideas?

**Comment by Iodine98** [comment]
> > How do we approach auditing code quality in the pipeline?
> 
> Apart from running `mllint`, which runs Code Quality (Pylint, Mypy, Black, isort, Bandit), but also the version control (Git and DVC), I am not sure how else we can audit code quality in the pipeline. Any ideas?
> 


Could we run `mllint` on every build of a version?

File: dvc.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+stages:
+  get_data:
+    cmd: python src/get_data.py
```
Comment (Iodine98) [inline dvc.yaml:3]: Shouldn't this be `poetry run python src/get_data.py`  to ensure that all Python scripts are executed within the defined Poetry environment? @rkargul

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+[tool.poetry]
+name = "model-training"
+version= "0.0.0"
+description = "This is the model training package"
+authors = [
+    "Radek Kargul <r.kargul@student.tudelft.nl>"
```
Comment (Iodine98) [inline pyproject.toml:6]: Add our names too pls:

"Rohan Sobha <r.r.sobha@student.tudelft.nl>"

File: src/preprocess_data.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# Load the data from the file
+print(os.getcwd())
+dataset = pd.read_csv("output/dataset.csv")
+nltk.download("stopwords")
+ps = PorterStemmer()
+all_stopwords = stopwords.words("english")
+all_stopwords.remove("not")
+
+corpus = []
+for i in range(0, 900):
```
Comment (Iodine98) [inline src/preprocess_data.py:20]: Are magic numbers allowed?

File: src/train.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.model_selection import train_test_split
+from sklearn.naive_bayes import GaussianNB
+
+# load the preprocessed data
+corpus = load("output/preprocessed_data.joblib")
+dataset = pd.read_csv("output/dataset.csv")
+
+# Data transformation
+cv = CountVectorizer(max_features=1420)
```
Comment (Iodine98) [inline src/train.py:13]: I guess this has been copied from Luis' project, right?

Comment (rkargul) [reply src/train.py:13]: Yes the model itself and its training was just taken from their project

**Review by Iodine98 — APPROVED** [review]
The changes have been implemented and my questions are answered. So I will approve of this PR

**Pull request merged**